modules/lang-painless/src/test/java/org/elasticsearch/painless/RegexTests.java

```diff
@@ -125,6 +126,51 @@ public void testNamedGroup() {
         assertEquals("o", exec("Matcher m = /(?<first>f)(?<second>o)o/.matcher('foo'); m.find(); return m.namedGroup('second')"));
     }
 
+    public void testMultilineFlag() {
+        assertEquals(false, exec("'a\nb\nc' =~ /^b$/"));
+        assertEquals(true, exec("'a\nb\nc' =~ /^b$/m"));
+    }
```

Comment:
also static method `matches` in Pattern is no good. otherwise it all seems fine.